cli/unstable_prompt_multiple_select.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+/** Options for {@linkcode promptMultipleSelect}. */
+export interface PromptMultipleSelectOptions {
+  /** Clear the lines after the user's input. */
+  clear?: boolean;
+}
+
+const ETX = "\x03";
+const ARROW_UP = "\u001B[A";
+const ARROW_DOWN = "\u001B[B";
+const CR = "\r";
+const INDICATOR = "❯";
+const PADDING = " ".repeat(INDICATOR.length);
+
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+const CLR_ALL = encoder.encode("\x1b[J"); // Clear all lines after cursor
+const HIDE_CURSOR = encoder.encode("\x1b[?25l");
+const SHOW_CURSOR = encoder.encode("\x1b[?25h");
+
+/**
+ * Shows the given message and waits for the user's input. Returns the user's selected value as string.
+ *
+ * @param message The prompt message to show to the user.
+ * @param values The values for the prompt.
+ * @param options The options for the prompt.
+ * @returns The selected values as an array of strings.
+ *
+ * @example Usage
+ * ```ts ignore
+ * import { promptMultipleSelect } from "@std/cli/prompt-select";
+ *
+ * const browsers = promptMultipleSelect("Please select browsers:", ["safari", "chrome", "firefox"], { clear: true });
+ * ```
+ */
+export function promptMultipleSelect(
+  message: string,
+  values: string[],
+  { clear }: PromptMultipleSelectOptions = {},
+): string[] {
+  const length = values.length;
+  let selectedIndex = 0;
+  const selectedIndexes = new Set<number>();
+
+  Deno.stdin.setRaw(true);
+  Deno.stdout.writeSync(HIDE_CURSOR);
+  const buffer = new Uint8Array(4);
+  loop:
+  while (true) {
+    Deno.stdout.writeSync(encoder.encode(`${message}\r\n`));
+    for (const [index, value] of values.entries()) {
+      const selected = index === selectedIndex;
+      const start = selected ? INDICATOR : PADDING;
+      const checked = selectedIndexes.has(index);
+      const box = checked ? "☑️" : "☐";
```

Comment:
These characters seem unbalanced in terminal to me

<img width="345" alt="Screenshot 2024-12-11 at 15 32 05" src="https://github.com/user-attachments/assets/0c62eb6a-9ad1-4eb0-9d23-4fbacfe24890">

How about using `◯` and `◉` which are used by `npm:prompts` and `npm:@inquirer/prompts` in a similar context.

`npm:prompts`
<img width="558" alt="Screenshot 2024-12-11 at 15 36 30" src="https://github.com/user-attachments/assets/a53afc5a-8878-4856-90da-411f079d2475">

`npm:@inquirer/prompts`
<img width="586" alt="Screenshot 2024-12-11 at 15 37 18" src="https://github.com/user-attachments/assets/bc210a27-0308-4053-a6eb-872d5f09ef19">

Reply:
Note: the code used for checking `npm:prompts` and `npm:@inquirer/prompts`

```js
import prompts from "npm:prompts";

await prompts({
  type: "multiselect",
  name: 'color',
  message: 'Pick colors',
  choices: [
    { title: 'Red', value: '#ff0000' },
    { title: 'Green', value: '#00ff00' },
    { title: 'Blue', value: '#0000ff' }
  ],
})
```

```js
import { checkbox, Separator } from "npm:@inquirer/prompts";

const answer = await checkbox({
  message: 'Select a package manager',
  choices: [
    { name: 'npm', value: 'npm' },
    { name: 'yarn', value: 'yarn' },
    new Separator(),
    { name: 'pnpm', value: 'pnpm', disabled: true },
    {
      name: 'pnpm',
      value: 'pnpm',
      disabled: '(pnpm is not available)',
    },
  ],
});
```

Reply:
Seems like the current checkboxes do not look the same in every terminal. I changed them.